packages/react/src/primitives/shared/responsive/__tests__/useBreakpoint.test.ts

```diff
@@ -0,0 +1,97 @@
+/** @jest-environment jsdom */
+
+import { useBreakpoint } from '../useBreakpoint';
+import { act, renderHook } from '@testing-library/react-hooks';
+import MatchMediaMock from 'jest-matchmedia-mock';
+import * as React from 'react';
+import { Breakpoints } from '../../../types/responsive';
+import { getMediaQueries } from '../getMediaQueries';
+
+jest.mock('react', () => ({
+  ...(jest.requireActual('react') as typeof React),
+  useDebugValue: jest.fn(),
+}));
+const mockUseDebugValue = React.useDebugValue as jest.Mock<
+  typeof React.useDebugValue
+>;
+
+const breakpoints: Breakpoints = {
+  xl: 80,
+  small: 30,
+  base: 0,
+  large: 62,
+  medium: 48,
+  xxl: 96,
+};
+
+const breakpointUnit = 'em';
+const defaultBreakpoint = 'base';
+
+let matchMedia: MatchMediaMock;
+let mediaQueries = getMediaQueries({ breakpointUnit: 'em', breakpoints });
+describe('useBreakpoint', () => {
+  beforeAll(() => {
+    matchMedia = new MatchMediaMock();
+  });
+
+  afterEach(() => {
+    matchMedia.clear();
+    mockUseDebugValue.mockClear();
+  });
+
+  it('should return default breakpoint', () => {
+    const { result } = renderHook(() =>
+      useBreakpoint({ breakpoints, breakpointUnit, defaultBreakpoint })
+    );
+    const breakpoint = result.current;
+
+    expect(breakpoint).toBe('base');
+  });
+
+  mediaQueries.forEach((m) => {
+    it(`should return ${m.breakpoint.toString()} breakpoint`, () => {
+      matchMedia.useMediaQuery(m.query);
+
+      const { result } = renderHook(() =>
+        useBreakpoint({ breakpoints, breakpointUnit, defaultBreakpoint })
+      );
+      const breakpoint = result.current;
+
+      expect(breakpoint).toBe(m.breakpoint);
+      expect(mockUseDebugValue).toHaveBeenLastCalledWith(
+        m.breakpoint,
+        expect.any(Function)
+      );
+      if (m.breakpoint === 'base') {
+        expect(mockUseDebugValue).toHaveBeenCalledTimes(1);
+      } else {
+        expect(mockUseDebugValue).toHaveBeenCalledTimes(2);
+      }
+    });
+  });
+
+  it('should react to changes to matchMedia', () => {
```

Comment:
The main difference is that the `should return ${m.breakpoint} breakpoint` tests only in one direction, but this tests going from up breakpoints, and then back down the breakpoints (base -> small -> medium -> small). This test ensures that the responsive support works when resizing the browser window wider, and then making it smaller again.

Reply:
However, it would probably be cleaner to reverse `mediaQueries`, and repeat the `should return ${m.breakpoint}` test to ensure it works both ways. Will update.